src/Controller/GameController.php

```diff
@@ -41,38 +43,81 @@ public function newGame(): Response
                 "id" => $game->getId()
             ]
         );
-
         $response->headers->setCookie(new Cookie($this::COOKIE_KEY, $playerHash));
 
         return $response;
     }
 
     /**
-     * @Route("/game/{id}", name="game")
-     * Used to display the board
+     * @Route("/game/{id}/invite", name="inviteGame")
      */
-    public function game(Request $request, string $id): Response
+    public function inviteGame(Request $request, string $id): Response
     {
         $playerHash = $request->cookies->get($this::COOKIE_KEY);
 
         $entityManager = $this->getDoctrine()->getManager();
+
         $game = $entityManager->getRepository(Game::class)->find($id);
+        if (!$game) {
+            return $this->redirectToRoute('newGame');
+        }
 
+        if (!$playerHash) {
+            $playerHash = $this->playerService->generatePlayerHash();
+        }
 
+        if ($game->getPlayer1Hash() === null) {
+            // If this game has no player 1 (this case is impossible in theory but for debugging purpose it's useful)
+            // Set this player as player 1
+            $game->setPlayer1Hash($playerHash);
+        } elseif ($playerHash !== $game->getPlayer1Hash() && $game->getPlayer2Hash() === null) {
+            // If game has no player 2
+            // Start this game !
+            $game = $this->gameService->setPlayer2AndStartGame($playerHash, $game);
+        }
 
-        if (!$playerHash || !$game) {
+        $entityManager->flush();
```

Comment:
The spectator is doing to much on this function. You need to generate a hash and flush the entity in spectator case?